File: CSharp/SpaceVIL/SpaceVIL/Cores/InputDeviceEvent.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SpaceVIL
{
    public class InputDeviceEvent
    {
        internal bool isHappen = false;
        private InputEventType EType = 0;
        public void SetEvent(InputEventType type)
        {
            EType |= type;
            isHappen = true;
        }
        public bool IsEvent()
        {
            bool tmp = isHappen;
            isHappen = false;
            return tmp;
        }
        public InputEventType LastEvent()
        {
            return EType;
        }
        public void ResetEvent(InputEventType type)
        {
            EType &= ~type;
        }
        public void ResetAllEvents()
        {
            EType = 0;
        }
    }
}

File: spacevil_21052019/java_src/InputDeviceEvent.java
package com.spvessel.spacevil;

import com.spvessel.spacevil.Flags.InputEventType;

import java.util.LinkedList;
import java.util.List;

final class InputDeviceEvent {
    Boolean isHappen = false;
    private List<InputEventType> eType = new LinkedList<>();

    void setEvent(InputEventType type) {
        if (!eType.contains(type))
            eType.add(type);

        isHappen = true;
    }

    Boolean isEvent() {
        Boolean tmp = isHappen;
        isHappen = false;
        return tmp;
    }

    List<InputEventType> lastEvent() {
        return eType;
    }

    void resetEvent(InputEventType type) {
        if (eType.contains(type))
            eType.remove(type);
    }

    void resetAllEvents() {
        eType.clear();
    }
}